compiler/haskell-ide-core/src/Development/IDE/State/FileStore.hs

```diff
@@ -141,7 +142,13 @@ setBufferModified state absFile (mcontents, !time) = do
     -- update vars synchronously
     modifyVar_ envDirtyFiles $ evaluate . case mcontents of
         Nothing -> Map.delete absFile
-        Just contents -> Map.insert absFile $ strictPair time (textToStringBuffer contents)
+        Just contents -> Map.insert absFile $ strictPair time $ textToStringBuffer contents
 
     -- run shake to update results regarding the files of interest
     void $ shakeRun state []
+
+
+-- would be nice to do this more efficiently...
+textToStringBuffer :: T.Text -> SB.StringBuffer
+-- would be nice to do this more efficiently...
```

Comment:
You must be very worried about efficiency here. ;)

Reply:
🤦‍♂

Reply:
Removed in https://github.com/digital-asset/daml/pull/963/files#diff-b8ba0a4cf59ff60c866c80c7ccb8f198